addons/l10n_in/models/account_invoice.py

```diff
@@ -87,7 +87,7 @@ def _post(self, soft=True):
         for move in posted.filtered(lambda m: m.country_code == 'IN'):
             if move.l10n_in_state_id and not move.l10n_in_state_id.l10n_in_tin:
                 raise UserError(_("Please set a valid TIN Number on the Place of Supply %s", move.l10n_in_state_id.name))
-            if not move.company_id.state_id:
+            if not (move.company_id.state_id or self._context.get('is_pos')):
```

Comment:
I don't really understand the context stuff ? Is it really needed ?

Reply:
Remove context.